notebook/agentchat_mongodb_RetrieveChat.ipynb

```diff
@@ -0,0 +1,538 @@
+     "output_type": "stream",
+     "text": [
+      "2024-06-22 21:19:19,131 WARNING autogen.oai.client: The API key specified is not a valid OpenAI format; it won't work with the OpenAI-hosted model.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# 1. create an RetrieveAssistantAgent instance named \"assistant\"\n",
+    "assistant = RetrieveAssistantAgent(\n",
+    "    name=\"assistant\",\n",
+    "    system_message=\"You are a helpful assistant.\",\n",
+    "    llm_config={\n",
+    "        \"timeout\": 600,\n",
+    "        \"cache_seed\": 42,\n",
+    "        \"config_list\": config_list,\n",
+    "    },\n",
+    ")\n",
+    "\n",
+    "# 2. create the RetrieveUserProxyAgent instance named \"ragproxyagent\"\n",
+    "# By default, the human_input_mode is \"ALWAYS\", which means the agent will ask for human input at every step. We set it to \"NEVER\" here.\n",
+    "# `docs_path` is the path to the docs directory. It can also be the path to a single file, or the url to a single file. By default,\n",
+    "# it is set to None, which works only if the collection is already created.\n",
+    "# `task` indicates the kind of task we're working on. In this example, it's a `code` task.\n",
+    "# `chunk_token_size` is the chunk token size for the retrieve chat. By default, it is set to `max_tokens * 0.6`, here we set it to 2000.\n",
+    "# `custom_text_types` is a list of file types to be processed. Default is `autogen.retrieve_utils.TEXT_FORMATS`.\n",
+    "# This only applies to files under the directories in `docs_path`. Explicitly included files and urls will be chunked regardless of their types.\n",
+    "# In this example, we set it to [\"non-existent-type\"] to only process markdown files. Since no \"non-existent-type\" files are included in the `websit/docs`,\n",
+    "# no files there will be processed. However, the explicitly included urls will still be processed.\n",
+    "ragproxyagent = RetrieveUserProxyAgent(\n",
+    "    name=\"ragproxyagent\",\n",
+    "    human_input_mode=\"NEVER\",\n",
+    "    max_consecutive_auto_reply=3,\n",
+    "    retrieve_config={\n",
+    "        \"task\": \"code\",\n",
+    "        \"docs_path\": [\n",
+    "            \"https://raw.githubusercontent.com/microsoft/FLAML/main/website/docs/Examples/Integrate%20-%20Spark.md\",\n",
+    "            \"https://raw.githubusercontent.com/microsoft/FLAML/main/website/docs/Research.md\",\n",
+    "            os.path.join(os.path.abspath(\"\"), \"..\", \"website\", \"docs\"),\n",
+    "        ],\n",
+    "        \"custom_text_types\": [\"non-existent-type\"],\n",
+    "        \"chunk_token_size\": 2000,\n",
+    "        \"model\": config_list[0][\"model\"],\n",
+    "        \"vector_db\": \"mongodb\",  # MongoDB Atlas database\n",
+    "        \"collection_name\": \"flaml_collection\",\n",
+    "        \"db_config\": {\n",
+    "            \"connection_string\": \"<connection_string>\",  # MongoDB Atlas connection string\n",
```

Comment:
Can you run the notebook with a mongodb instance deployed with the docker-compose.yml provided at the beginning and share the actual connection_string you used?